packages/fluentui/react-northstar/src/components/Datepicker/DatepickerCalendar.tsx

```diff
@@ -0,0 +1,202 @@
+import { commonPropTypes, createShorthandFactory, UIComponentProps } from '../../utils';
+import Grid from '../Grid/Grid';
+import {
+  IDayGridOptions,
+  getDayGrid,
+  IDay,
+  DAYS_IN_WEEK,
+  IDateGridStrings,
+  DayOfWeek,
+  FirstWeekOfYear,
+  DateRangeType,
+  formatMonthDayYear,
+} from '@fluentui/date-time-utilities';
+import Text from '../Text/Text';
+import Button from '../Button/Button';
+import { IDatepickerOptions } from './Datepicker';
+
+export interface IDateFormatting {
+  /**
+   * Format the date according to specified function.
+   * Intended use case is localization.
+   */
+  format?: (date: Date) => string;
+
+  /**
+   * Parse date from string representation into Date type.
+   */
+  parse?: (date: string) => Date;
+}
+
+export interface DatepickerCalendarProps extends IDatepickerOptions, IDateFormatting, UIComponentProps {
+  /**
+   * The currently selected date
+   */
+  selectedDate: Date;
+  /**
+   * The currently navigated date
+   */
+  navigatedDate: Date;
+  /** Accessibility behavior if overridden by the user. */
+  accessibility?: Accessibility<DatepickerCalendarBehaviorProps>;
+
+  /**
+   * Called on change of the date.
+   *
+   * @param event - React's original SyntheticEvent.
+   * @param data - All props and proposed value.
+   */
+  onDaySelect?: (e: React.MouseEvent<HTMLElement>, data: DatepickerCalendarProps & { value: IDay }) => void;
+
+  /** Localized labels */
+  localizedStrings?: IDateGridStrings;
+}
+
+export type DatepickerCalendarStylesProps = never;
+
+export const datepickerCalendarClassName = 'ui-datepickerCalendar';
+
+const DatepickerCalendar: ComponentWithAs<'div', DatepickerCalendarProps> &
+  FluentComponentStaticProps<DatepickerCalendarProps> = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(DatepickerCalendar.displayName, context.telemetry);
+  setStart();
+  const datepickerCalendarRef = React.useRef<HTMLElement>();
+  const {
+    className,
+    design,
+    styles,
+    variables,
+    selectedDate,
+    navigatedDate,
+    firstDayOfWeek,
+    firstWeekOfYear,
+    dateRangeType,
+    localizedStrings,
+  } = props;
+  const ElementType = getElementType(props);
+  const unhandledProps = useUnhandledProps(DatepickerCalendar.handledProps, props);
+  const getA11yProps = useAccessibility(props.accessibility, {
+    debugName: DatepickerCalendar.displayName,
+    actionHandlers: {},
+    rtl: context.rtl,
+  });
+  const gridOptions: IDayGridOptions = {
+    selectedDate,
+    navigatedDate,
+    firstDayOfWeek,
+    firstWeekOfYear,
+    dateRangeType,
+  };
+
+  const { classes } = useStyles<DatepickerCalendarStylesProps>(DatepickerCalendar.displayName, {
+    className: datepickerCalendarClassName,
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  });
+
+  let grid = getDayGrid(gridOptions);
```

Comment:
I don't like that we are randomly using `date` vs `day`. Should we stick with one of them? If it is already like that on v7 disregard this comment, but it may be a good thing to be looked over..

Reply:
In general my idea was to differentiate between day, month and year pickers, so now we are getting dayGrid, and then for month picker it would be getting monthGrid or something like that. So, I'd leave this like that for now :)